database/sql/headers.go

```diff
@@ -187,11 +195,30 @@ func (h *HeadersDb) Create(ctx context.Context, req dto.DbBlockHeader) error {
 		_ = tx.Rollback()
 	}()
 	if _, err := tx.NamedExecContext(ctx, sqlInsertHeader, req); err != nil {
+		if h.isUniqueViolation(err) {
+			return customErrs.NewUniqueViolationError()
+		}
+
 		return errors.Wrap(err, "failed to insert header")
 	}
+
 	return errors.Wrap(tx.Commit(), "failed to commit tx")
 }
 
+func (h *HeadersDb) isUniqueViolation(err error) bool {
+	switch h.db.DriverName() {
```

Comment:
Here I'm thinking about type switch between `sqlite3.Error` and `pq.Error`. In this approach, the DriverName would not be necessary.

Reply:
fixed - replaced switch DriverName with switch err(type)